### PR TITLE
Weekend bug hunt 🦠

### PR DESCRIPTION
**Pull request by Nkluge-correa**
- Fixed some bugs regarding the GRPO implementation (imports and the parsing of reasoning traces). Update the tests to catch these bugs in the future
- Added info to the documentation/comments in alignment training scripts
- Simplified the support for non-softmax-attention architectural implementations. Using qwen3.5 as a reference (mostly to not lose the support of liger kernels)
- Fixed a small bug in the math verifier (now it is more robust when working with fractions).
- Added `upload_quick.py` to `./utils`. This helps to push/correct files in all branches of an HF repo

### CODE REVIEW

**Comment by Nkluge-correa** [comment]
🪰
